cdap-watchdog/src/main/java/co/cask/cdap/logging/clean/FileMetadataCleaner.java

```diff
@@ -61,11 +61,10 @@ public FileMetadataCleaner(TransactionRunner transactionRunner) {
 
         boolean reachedEnd = false;
         Range range = Range.all();
-        int count = 0;
         // stop cleaning up if there are no files to delete or if we have reached batch size
-        while (!reachedEnd && count < fileCleanupBatchSize) {
-          try (CloseableIterator<StructuredRow> iter = table.scan(range, Integer.MAX_VALUE)) {
-            while (count < fileCleanupBatchSize) {
+        while (!reachedEnd && toDelete.size() < fileCleanupBatchSize) {
```

Comment:
Let's move the starting of new transaction inside the while loop so that every scan is in a new transaction. This will prevent transaction timeout in the NoSQL case.